components/create_team/create_team.tsx

```diff
@@ -1,68 +1,76 @@
 import React from 'react';
-import {Route, Switch, Redirect} from 'react-router-dom';
+import {Route, Switch, Redirect, RouteComponentProps} from 'react-router-dom';
+
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {Team} from 'mattermost-redux/types/teams';
 
 import AnnouncementBar from 'components/announcement_bar';
 import BackButton from 'components/common/back_button';
 import DisplayName from 'components/create_team/components/display_name';
 import SiteNameAndDescription from 'components/common/site_name_and_description';
 import TeamUrl from 'components/create_team/components/team_url';
 
-export default class CreateTeam extends React.PureComponent {
-    static propTypes = {
-
-        /*
-         * Object containing information on the current team, used to define BackButton's url
-         */
-        currentTeam: PropTypes.object,
-
-        /*
-         * Object containing information on the current selected channel, used to define BackButton's url
-         */
-        currentChannel: PropTypes.object,
-
-        /*
-         * String containing the custom branding's text
-         */
-        customDescriptionText: PropTypes.string,
-
-        /*
-         * String containing the custom branding's Site Name
-         */
-        siteName: PropTypes.string,
-
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            url: PropTypes.string.isRequired,
-        }).isRequired,
-    }
+type Props = {
+
+    /*
+   * Object containing information on the current team, used to define BackButton's url
+   */
+    currentTeam: Team;
+
+    /*
+   * Object containing information on the current selected channel, used to define BackButton's url
+   */
+    currentChannel: Channel;
 
-    constructor(props) {
+    /*
+    * String containing the custom branding's text
+    */
+    customDescriptionText: string;
+
+    /*
+   * String containing the custom branding's Site Name
+   */
+    siteName: string;
+
+    /*
+   * Object from react-router
+   */
+    match: {
+        url: string;
+    };
+};
+
+type State = {
+    team: object;
+    wizard: string;
+};
+
+export default class CreateTeam extends React.PureComponent<Props & RouteComponentProps<{}>, State> {
+    public constructor(props: Props & RouteComponentProps<{}>) {
         super(props);
 
-        const state = {};
-        state.team = {};
-        state.wizard = 'display_name';
-        this.state = state;
+        this.state = {
+            team: {},
+            wizard: 'display_name',
+        };
     }
 
-    updateParent = (state) => {
+    public updateParent = (state: State) => {
         this.setState(state);
         this.props.history.push('/create_team/' + state.wizard);
-    }
+    };
 
     render() {
         const {
             currentChannel,
             currentTeam,
             customDescriptionText,
             match,
-            siteName,
+            siteName
```

Comment:
Looks like our eslint rules are a bit out of whack at the moment. As far as I'm aware, we expect a trailing comma here like so:
```suggestion
            siteName,
```